### ISSUE TITLE
% ceded does not work on QS or CAT XL

### ISSUE THREAD

**Issue by aiste-kalinauskaite** [OasisLMF/ReinsuranceTestTool#26]
in the bottom 8 examples, changing % ceded, does not change net losses, although I would expect them to change, as the proportion ceded to reinsurer would vary.

**Comment by johcarter** [comment]
Just tried changing % ceded for simple_QS and it works. Is it possible that you made the change in the grid and then reran the step to view ri_info? This would reload the original file and lose your edit.
Alternative explanation may be that you didnt rerun step 7 to update your change to the file?
Please can you try again and let me know?

**Comment by aiste-kalinauskaite** [comment]
This relates to the fact, that I was expecting all limits to be 100% and therefore % ceded (or rather after we did the poll, it will be % placed) would impact the losses after the limit (referencing Matt's email from Wednesday, it will no longer be that). Based on the poll, however, we need the flexibility, so there will be scenarios, were limits (and attachments?) will be represented as a share of the treaty rather than 100%. To differentiate which one it is, % ceded will be used with treaty share limits, while % placed will be used with 100% limit. so whether % ceded or % placed is populated (and everything else stays the same) - these will give very different answers. As per Matt's email, % ceded applies to gross losses, but % placed applies, post treaty limits and attachments.

For CAT XL, I believe we are anticipating only % placed to be used (so % ceded stays at 100%). Otherwise, I find it rather confusing, as to what do you do with OccAttachments. Are they scaled to treaty share too? 

For QS, both ways would be allowed. When % ceded and limits are populated, then those limits represent treaty share limits. But if % placed is used (and % ceded then is 100%), then the limits will be 100% limits. Although, I personally find 100% limits much easier to understand, apparently, they do not have to be expressed that way. Therefore we are allowing both.

With those changes in mind, I think QS might be ok (if the limits are treaty share limits and not 100% as I thought), but I will retest. As CAT XL, if attachment also needs to be scaled to treaty share, then I don't think it's working correctly. But as I said, I am not sure we need this case (with % ceded) and I can't test with % placed.

**Comment by johcarter** [comment]
If you could develop some test cases please that would be really handy

**Comment by aiste-kalinauskaite** [comment]
QS with % ceded and limits expressed as treaty shares work correctly. I can't test the other case, as % placed is disabled.

**Comment by aiste-kalinauskaite** [comment]
this is how CATXL works with % placed:
20% on 350 xs 50 layer should be coded as:
occLimit = 350
OccAttachment = 50
% placed = 0.2

This means that the maximum you would ever cede to reinsurer is OccLimit * % placed, i.e. 350 * 0.2 = 70. So if you had incoming loss as 400 (or higher), then you would always cede 70 and your net would be incoming loss - 70 back-allocated between locations.
If your incoming loss is < 50, then you would cede nothing to reinsurer and gross loss = net loss (as reinsurance program does not kick in).
If say, your incoming loss is 100, then you cede (100-50)*0.2 = 10 to reinsurer. 

Principal for QS with % placed is the same - you do per risk limits first, then occ limit, then apply % placed. Hope this helps, but I can go through this on Monday, if you need. It may well be that this already works correctly, so if we get % placed enabled, I can check.

**Comment by johcarter** [comment]
QS % ceded is working now. 
CAT XL % ceded working now.
Also % placed implemented
In v0.4 (dev)
@aiste-kalinauskaite please can you retest and close the issue if happy? thanks

**Comment by aiste-kalinauskaite** [comment]
% ceded and % placed seems ok, apart from a particular case with multiple CAT XLs, I raised as a separate issue.
